.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,308 @@
+### Table of Contents
+1. [Overview](#Overview)<br />
+2. [Minimum Recommended Skillset](#MinSkillset)<br />
+3. [Required Reading](#ReqReading)<br />
+4. [Development Practices](#DevelopmentPractices)<br />
+4.1. [Share Early, Share Often](#ShareEarly)<br />
+4.2. [Common Mistakes](#CommonMistakes)<br />
+4.3. [Testing](#Testing)<br />
+4.4. [Code Documentation and Commenting](#CodeDocumentation)<br />
+4.5. [Model Git Commit Messages](#ModelGitCommitMessages)<br />
+5. [Code Approval Process](#CodeApproval)<br />
+5.1 [Code Review](#CodeReview)<br />
+5.2 [Rework Code (if needed)](#CodeRework)<br />
+5.3 [Acceptance](#CodeAcceptance)<br />
+6. [Contribution Standards](#Standards)<br />
+6.1. [Contribution Checklist](#Checklist)<br />
+6.2. [Licensing of Contributions](#Licensing)<br />
+
+<a name="Overview" />
+
+### 1. Overview
+
+Developing cryptocurrencies is an exciting endeavor that touches a wide variety
+of areas such as wire protocols, peer-to-peer networking, databases,
+cryptography, language interpretation (transaction scripts), RPC, and
+websockets.  They also represent a radical shift to the current fiscal system
+and as a result provide an opportunity to help reshape the entire financial
+system.  There are few projects that offer this level of diversity and impact
+all in one code base.
+
+However, as exciting as it is, one must keep in mind that cryptocurrencies
+represent real money and introducing bugs and security vulnerabilities can have
+far more dire consequences than in typical projects where having a small bug is
+minimal by comparison.  In the world of cryptocurrencies, even the smallest bug
+in the wrong area can cost people a significant amount of money.  For this
+reason, bcoin has a formalized and rigorous development process which
+is outlined on this page.
+
+We highly encourage code contributions, however it is imperative that you adhere
+to the guidelines established on this page.
+
+<a name="MinSkillset" />
+
+### 2. Minimum Recommended Skillset
+
+The following list is a set of core competencies that we recommend you possess
+before you really start attempting to contribute code to the project.  These are
+not hard requirements as we will gladly accept code contributions as long as
+they follow the guidelines set forth on this page.  That said, if you don't have
+the following basic qualifications you will likely find it quite difficult to
+contribute.
+
+- A reasonable understanding of bitcoin at a high level (see the
+  [Required Reading](#ReqReading) section for the original white paper)
+- Experience in some type of C-like language
```

Comment:
for specifics to our codebase it might be good to also highlight NodeJs and ES6+/Modern JS syntax (e.g. JS classes, async/await, etc.). bcoin is quite unique in how up to date it is and we will definitely be trying to enforce those standards rigorously.